external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -133,6 +137,10 @@ private[kafka010] class KafkaMicroBatchStream(
 
   override def toString(): String = s"KafkaV2[$kafkaOffsetReader]"
 
+  override def metrics(latestOffset: Optional[Offset]): ju.Map[String, String] = {
+    KafkaMicroBatchStream.metrics(latestOffset, latestPartitionOffsets)
```

Comment:
I think it's ok to directly override metrics instead of having the new companion object `KafkaMicroBatchStream`

Reply:
Never mind, the object is added for testing I think.

Reply:
Yes. It's to unit test the calculation.